src/application.rs

```diff
@@ -82,11 +82,92 @@ pub trait ApplicationHandler<T: 'static = ()> {
     /// [`Suspended`]: Self::suspended
     fn resumed(&mut self, event_loop: &ActiveEventLoop);
 
-    /// Emitted when an event is sent from [`EventLoopProxy::send_event`].
-    ///
-    /// [`EventLoopProxy::send_event`]: crate::event_loop::EventLoopProxy::send_event
-    fn user_event(&mut self, event_loop: &ActiveEventLoop, event: T) {
-        let _ = (event_loop, event);
+    /// Called after a wake up is requested using [`EventLoopProxy::wake_up`].
+    ///
+    /// Multiple calls to the aforementioned method may result in only a single `proxy_wake_up` call
+    /// for performance reasons. Furthermore, this callback may be emitted spuriously. For these
```

Comment:
I deliberately kept the language vague here, in particular the note about spurious wakeups is mostly to discourage the user from thinking that they can do "receiver.try_recv().unwrap()", and expect that to always work.

Also, I didn't really want to promise more than necessary, but if "no spurious wakeups" is a must, then I'm fine with writing it differently - maybe you could suggest a wording?

Reply:
> Also, I didn't really want to promise more than necessary, but if "no spurious wakeups" is a desired thing, ...

I don't think its really desirable because as you say, users still have to account for `receiver.try_recv().unwrap()` not always work, which looks indeed like a spurious wake up. But if we explain to users what we are actually doing, I think we can tighten the expectations.

> Multiple calls to the aforementioned method will be merged and only wake up the event loop once. Due to the nature of multi-threading some wake ups may appear spuriously.

Reply:
Yeah, that's better, thanks!

Reply:
Thanks for the improvements on the wording, its great!